### ISSUE TITLE
Reading: Training Data Extraction From Pre-trained Language Models: A Survey

### ISSUE THREAD

**Issue by ayaco0** [ayaco0/paper-survey#22]
### 論文リンク
https://arxiv.org/pdf/2305.16157.pdf
### 著者/所属機関
Shotaro Ishihara / Nikkei Inc.
### 投稿年
2023
## 研究背景
- 統計的モデルから大規模データセットを用いた事前学習済み言語モデル（PLM）が主流に
- PLMの発展の一方で学習データ抽出による情報漏洩といった問題がある
- PLMに対するセキュリティ技術はまだまだ未発達

## サーベイ方法
- 100以上の研究をサーベイ
- Carlini et al.(2021) を引用している先行研究を関係性、引用数、アクセプトの有無を基準にスクリーニング

### 攻撃/防御のアプローチの体系化
<img width="600" alt="スクリーンショット 2023-06-21 11 52 54" src="https://github.com/ayaco0/paper-survey/assets/74520178/bcc742f0-2288-4ab2-ad89-2c0714187431">

#### 攻撃
- STEP1: Candidate generation
  - デコーディング手法の選択
  - プロンプトの選択
- STEP2: Membership inference
  - 機械学習モデルの学習に特定の例が含まれるか予測すること
  - 対象モデル：今回は自己回帰言語モデル
  - 攻撃者の知識
    - ホワイトボックス攻撃
      - 学習手順、モデルのアーキテクチャ、学習パラメータなど全ての情報を攻撃に使用できる
      - 訓練データが公開されている
    - ブラックボックス攻撃
      - full confidence scores：全ての出力が利用可能で損失の計算ができる
      - top-k confidence scores：出力の候補を複数個利用可能だが損失の計算はできないため攻撃の範囲が限定的
      - prediction labels only：ラベルのみ利用可能（DeepLやChatGPTなど）
      - 訓練データは非公開or一部の未公開
  - 手法
    - 分類器ベース、指標ベース、差分比較などの手法がある
    - メンバーシップ推論の指標にはパープレキシティが使われることが多い
  - アルゴリズム
    - 中央集権的 or 連合的なアルゴリズムがある
    - 連合学習はプライバシー保護的に良いが学習データ抽出研究では中央集権的手法が主流
  - ドメイン：臨床データ、契約データ、ソースコード、ニュース業界など

#### 防御
- STEP1：前処理
  - Data sanitization
    - 個人情報を含むテキストを特定して削除
    - プライバシーは文脈依存なため文字列からの特定は困難
  - Data deduplication
  - データ重複の削除がPLMの記憶を緩和する
  - モデルを訓練する方法より効率的
- STEP2：学習
  - Differential privacy
    - 差分プライパシーをPLMに適用
    - 性能低下やメモリ使用量・計算量の増加が課題
    - どこからか秘密かという境界を定義するのも課題 
  - Regularization
    - モデルの過学習は記憶と関係している（汎化されてたら個人情報は含まない？）
    - 情報ボトルネック層によって出力の表現に制約を与える研究も
    - ロスが低い学習データは出現率が高くなる？
    - 過学習前にほとんどのデータを記憶していて学習の過程で適切に忘れていく？
  - Knowledge distillation：知識上流
- STEP3：後処理
  - PLMの出力を制限
  - Confidence masking、Filteringなどがある

### 実証的な知見
- Larger models memorize more：モデルサイズが大きくなるほど記憶量が増加
  - Carlini et al. (2023b) ではモデルサイズが10倍になると暗記量は19ppt増加
- Duplicate strings are memorized：データの重複度が大きいほど記憶量が増加
- Longer prompts extract more：プロンプト長が長いほど記憶量が増加
  - ユーザのプロンプト長に制限を設けることで抽出リスクを低減できる？

### 今後の攻撃の方向性
- 記憶は悪か？
  - 抽出されるデータの危険性評価
    - 個人情報は抽出されたら危険だが、抽出されても問題のない情報もある
    - PLMは個人情報を特定の人に結びつけることができないため問題ない？ (Huang et al., 2022)
  - PLMの記憶と知識の境界は曖昧なまま
    - 記憶させないようにしすぎたら有用な知識も失う？
    - モデル性能とセキュリティのトレードオフが課題
-  研究分野の拡大に向けて
      - 英語以外の言語ではそれぞれ考慮すべきことが異なる可能性 (Ippolito et al., 2022)
      - **画像生成の記憶、剽窃検出**という研究領域も↑で提案
- 評価スキーマ
  - 評価セットの構築に工夫の余地あり
  - 推論時間の制約があるからサンプリングは避けられない（訓練データからのサンプリングが多い）
  - バイアスを考慮するためには、分布の重複以外にも注意するべき？
  - 評価指標にも議論の余地あり
    - 平均精度だけで良いのか？攻撃速度や再現率など？
    - 現実の攻撃シナリオに即した評価指標にしなきゃよね

## 感想
- [ozdayi et al.(2023) ](https://arxiv.org/abs/2305.11759) 読んで訓練データ抽出に興味が湧いたからこれ読みたくなった
- 石原さんがNLP2023でも訓練データ抽出に関しての発表してた

## 参考
-  [差分プライバシーの概要と機械学習への応用](https://www.skillupai.com/blog/tech/differential-privacy/)
